docs/command-line-interface/ds-load-cli/usage.mdx

```diff
@@ -0,0 +1,192 @@
+* `--host` is a CLI parameter for `ds-load`
+* `auth0` is the plugin name
+* `--domain` is a parameter for the `auth0` plugin
+
+For viewing the plugin help, use the following format: `ds-load auth0 --help`. 
+
+:::tip Tip
+When running `ds-load auth0 --key`, `auth0` is a positional parameter, so `--key` will be run in the context of the plugin. If we run `ds-load --key auth0`, `--key` would be a parameter to `ds-load exec`.
+:::
+
+### ds-load exec
+`exec` is the default command. it will invoke a plugin with the specified parameters reading its output and importing the resulting data into the directory.
+
+```sh
+Usage: ds-load exec <command> ...
+
+import data in directory by running fetch, transform and publish
+
+Arguments:
+  <command> ...    available commands are: auth0|azuread|cognito|google|okta
+
+Flags:
+  -h, --help                  Show context-sensitive help.
+  -c, --config=CONFIG-FLAG    Path to the config file. Any argument provided to the CLI will take precedence.
+  -v, --verbosity=INT         Use to increase output verbosity.
+
+  -s, --host=STRING           Directory host address ($DIRECTORY_HOST)
+  -k, --api-key=STRING        Directory API Key ($DIRECTORY_API_KEY)
+  -i, --insecure              Disable TLS verification
+  -t, --tenant-id=STRING      Directory Tenant ID ($DIRECTORY_TENANT_ID)
+  -p, --print                 print output to stdout
+```
+
+`-p/--print` is enabled by default when invoking a plugin with `fetch/version/export-transform/--help`
+
+### Environment variables
+Parameters can also be passed using environment variables, as seen in the help message of each command, but the ones from config files and command line take precedence.
+
+## Config files
+
+Config files are in yaml format:
+```yaml
+---
+arg: value
+another-arg: value
+<plugin-name>:
+  arg: value
+```
+
+When passing custom config files to both the cli and the plugin, use `ds-load -c <config-path> <plugin-name> <command>` 
+
+### CLI config
+The default location for the configuration file is `~/.config/ds-load/cfg/config.yaml`. It can be overridden using the `-c/--config` flag.
+
+#### example with auth0 plugin
+```yaml
+---
+host: directory.prod.aserto.com:8443
+api-key: secretapikey
+tenant-id: your-tenant-id
+auth0:
+  domain: "domain.auth0.com"
+  client-id: "clientid"
+  client-secret: "clientsupersecret"
+  template-file: "/path/to/transform.file"
+```
+
+### Plugin config
+The default location for plugin configuration files is `~/.config/ds-load/cfg/<plugin-name>.yaml`. It can be overridden using the `-c/--config` flag.
+
+#### example for auth0
+```yaml
+---
+auth0:
+  domain: "domain.auth0.com"
+  client-id: "clientid"
+  client-secret: "clientsupersecret"
+  template-file: "/path/to/transform.file"
+```
+
+## Transform
+The data received from the fetcher is transformed into objects and relations using a transformation template, which is uses the go template syntax.
```

Comment:
The fetcher is first introduced here, maybe we could give a short description of the pipeline first e.g.

ds-load has 3 main steps fetch, transform, and publish. in the first step, it retrieves raw information from the identity provider, the second step transforms this raw data into objects and relations and the third step saves the data in the directory.